Fix code snippet formatting alignment issues

michelle-purcell opened the pull request:
Minor edits to enhance ordering and alignment of optional choice steps and some other style quality edits.

_(Some of these edits are required for single-sourcing downstream docs)._

michelle-purcell commented:
In draft while I review the preview up and downstream.

MichalMaler commented on docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:
```diff
@@ -46,46 +46,61 @@ You can find the solution in the `security-jpa-quickstart` link:{quickstarts-tre
-After you run either of the preceding commands, verify that the `security-jpa` dependency was added to your project build XML file,as follows:
+After you have run either of the preceding commands to create the Maven project, verify that the `security-jpa` dependency was added to your project build XML file.
 
+* To verify the `security-jpa` extension, check for the following configuration:
```
Hello @michelle-purcell . This PR is another digital splendor :))
Here is just one little bit that needs to be addressed and you have two options:
1) Use `+` to glue these two snippets together to make them both properly align under the bullet

```
* TextText:
+
[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
.pom.xml
----
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-security-jpa</artifactId>
</dependency>
----
+
[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
.build.gradle
----
implementation("io.quarkus:quarkus-security-jpa")
----
```

2) Use `+` right after the bullet, then wrap the two snippets in `====` box. This will create an encapsulated unit that is glued to its sticking point :)

```
* TextText:
+
====
[source,xml,role="primary asciidoc-tabs-target-sync-cli asciidoc-tabs-target-sync-maven"]
.pom.xml
----
<dependency>
    <groupId>io.quarkus</groupId>
    <artifactId>quarkus-security-jpa</artifactId>
</dependency>
----

[source,gradle,role="secondary asciidoc-tabs-target-sync-gradle"]
.build.gradle
----
implementation("io.quarkus:quarkus-security-jpa")
----
====
```

I am horrible when it came to design, so I will keep this one up to you :).

MichalMaler commented on docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:
```diff
@@ -46,46 +46,61 @@ You can find the solution in the `security-jpa-quickstart` link:{quickstarts-tre
-* To create the Maven project with Hibernate ORM, use the following command:
+=== Create the Maven project
+
+.Create a new Maven project with the Jakarta Persistence extension
```
.Create a new Maven project with the Jakarta Persistence extension -> . Create a new Maven project with the Jakarta Persistence extension

If those should be numbered steps in the end.

MichalMaler commented on docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:
```diff
@@ -46,46 +46,61 @@ You can find the solution in the `security-jpa-quickstart` link:{quickstarts-tre
-You must also add your preferred database connector library.
-The instructions in this example tutorial use a PostgreSQL database for the identity store.
-====
+.Add the Jakarta Persistence extension to an existing Maven project
```
.Add the Jakarta Persistence extension to an existing Maven project -> . Add the Jakarta Persistence extension to an existing Maven project

If this should be numbered step.

MichalMaler commented:
@michelle-purcell LGTM! Added some tips/suggestions. Have a look and apply if seems fit :)

michelle-purcell commented:
Verified that the code snippet steps look perfect in preview and are also now fixed downstream with this PR plus @MichalMaler 's downstream doc postfix code snippet script.

michelle-purcell commented:
@ebullient - As SB is on PTO, could you review and approve these changes? 
We are enhancing the way code snippets look downstream and this has highlighted some non-technical (from the Quarkus Security perspective)  markup/style issues. Thank you :pray:

michelle-purcell commented:
@ebullient - Thanks for approving :green_heart:

gsmet committed:
Fix code snippet formatting alignment issues

Fix code snippet formatting alignment issues

Fix optional step formatting

Added more code wrappers to align tutorial steps

Fix typos

Fix alignment

Fix typo

Minor fix

gsmet commented on docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:
```diff
@@ -72,20 +59,49 @@ You must also add your preferred database connector library.
+* To create a new Maven project with the Jakarta Persistence extension, complete one of the following steps:
+** To create the Maven project with Hibernate ORM, use the following command:
++
+====
```
I have no idea what this is in AsciiDoctor when you don't have a `[NOTE]`? I don't think we ever used that construct.
Does it bring something to the plate?

michelle-purcell replied:
@gsmet - Wrapping a numbered (. ) or bulleted (* ) step in a procedure (how-to/tutorial) by adding `====` to encapsulate code snippets fixes an alignment and numbering issues we are seeing both up and downstream.

For example, we were seeing '+' in the rendered output instead of a code snippet aligned and indented to match the current step, and the step numbering was broken or restarted back to *1.* again :
![image](https://github.com/quarkusio/quarkus/assets/92924207/3abab7fe-05f6-4607-a2cb-c3b97c414f4b)

By wrapping the code lines like so:
````
. The step 1 text goes here.
+
====
 code snippet include and attribute lines
====

. The next step goes here. If this has code, repeat and rinse:
+
====
code include etc
====

. The next step.... 
````
Glues all of the code lines for a step together and preserves the step numbering and indentation, for example:

![image](https://github.com/quarkusio/quarkus/assets/92924207/8765f680-34d1-4fb9-8129-2b7da5795859)

Does that make sense?

p.s my last commit introduced an asciidoc error. I am fixing now.... Thanks.

ebullient replied:
Should add something to reference or howto doc (separate PR is fine) about how/when to use this construct to ensure numbered lists function properly.

gsmet replied:
@michelle-purcell thanks for the explanation but I want to make sure you have tested this new layout on the website? We don't have everything AsciiDoctor implemented there so we need to make sure it looks good. Thanks for confirming.

gsmet reviewed (commented):
I added one question regarding an AsciiDoctor construct that I never saw us use.

If we absolutely have too, we will need to make sure it works well on the website too.

MichalMaler commented:
@gsmet Hello G., Just for the context; thats also why I asked you to add that trim of the empty lines from the includes. When there are no empty lines ,`+` works as a glue that sticks asciidoc pieces together. When there is an empty line in the include however, the `+` is not attached to the file as a unit, but to its last empty row, and breakes the asciidoc markup.
Since the trim is now part of your script, we should be good with this approach now :)
It works well in both Web portals;  it just introduces a new element to the final doc -> that ==== box around the wrapped content.

michelle-purcell commented:
@MichalMaler - Thanks so much for that explanation and for figuring out this workaround :100:  :+1:

ebullient commented on docs/src/main/asciidoc/security-basic-authentication-tutorial.adoc:
```diff
@@ -227,7 +254,7 @@ public class User extends PanacheEntity {
      * @param username the username
      * @param password the unencrypted password (it will be encrypted with bcrypt)
-     * @param role the comma-separated roles
+          * @param role the comma-separated roles
```
extra whitespace?

michelle-purcell replied:
Fixed. Thanks for flagging @ebullient . Must've elbowed the space bar! :mechanical_arm:

ebullient reviewed (approved):
nit whitespace change..

michelle-purcell commented:
@ebullient / @gsmet - Can we merge this PR? Thanks

ebullient commented:
double checking website rendering.. then I'll merge.

ebullient commented:
Yep -- @michelle-purcell snipped from website preview [here](https://github.com/quarkusio/quarkus/pull/35274#discussion_r1295838866) ... we're good.

michelle-purcell commented:
Thank you @ebullient :+1: 

 Just a heads up @ebullient, @sberyozkin, and @gsmet - I'll also need to apply the numbering alignment fixes to the other security tutorials and they will also need to be backported to 3.2 after being merged to the main branch.
 I'll do this soon. Thanks.

michelle-purcell commented:
@gsmet - Oops sorry! Didn't mean to read the 3.2 label again. I see you are in the process of backporting.

I was looking to see if there's such a thing as a 3.2.4 label? Are `3.2.4` docs on the `3.2` branch?
